### DISCUSSION TITLE
Lock File Maintenance PR not passing pipeline?

### DISCUSSION THREAD

**Discussion by dbimhoff** [renovatebot/renovate#10502] [Request Help] [answered]
Hello, I'm running a self hosted renovate on gitlab runners and I keep getting this Lock File Maintenance PR that fails to build and I'm not sure why. 
![image](https://user-images.githubusercontent.com/47460175/122593533-6ca73280-d02b-11eb-8f36-dff45b2d9df8.png)

In my config.js I have lockFileMaintenance enabled and it seems to be working on every other dependency PR update, so I'm not even sure why this one is needed if the lock file is already getting updated. Any insights?

Here is my config.js if that helps at all.

```JS
module.exports = {
    platform: 'gitlab',
    username: 'renovate-bot',
    gitAuthor: 'Renovate Bot <bot@example.com>',
    endpoint: '<removed for privacy>',
    assignees: [],
    baseBranches: ['master'],
    labels: ['renovate'],
    onboarding: false,
    requireConfig: false,
    dependencyDashboard: true,
    lockFileMaintenance: { 'enabled': true, 'schedule': 'at any time' },
    recreateClosed: true,
    repositories: [
    <removed for privacy>
    ],
    // extends: ["config:base"],
    npmrc: "registry=<removed for privacy>\nlegacy-peer-deps=true",

  };
```

**Comment by rarkins** [comment, marked as answer]
It will set a failing check if npm returns an error, so that you will be aware. Check your debug logs from when the commit was made to determine what went wrong in npm

**Reply by dbimhoff** [reply]
Do you know what I should be looking for exactly? I looked through and I didn't seem to find any error reports or anything

**Reply by rarkins** [reply]
Look for the exec commands where npm is called. Possibly you are using the default npm v6 in our published image while your repo needs v7?

**Reply by dbimhoff** [reply]
I'm pretty sure I am running npm 7. I use this custom dockerfile to build our image to use npm 7.
![image](https://user-images.githubusercontent.com/47460175/122609522-bd2a8a00-d043-11eb-93eb-fc6011d6abed.png)

**Reply by rarkins** [reply]
Look for the log `Executing command` at the time just prior to the commit. After that you should see some indication of the npm result, such as "rawExec err"

**Reply by dbimhoff** [reply]
Where do I see that? I don't see it on the logs on gitlab with debug level on

**Reply by viceice** [reply]
Which version of renovate are you using?

**Reply by dbimhoff** [reply]
it should be the most recent on docker hub I believe since we pull it from docker in our dockerfile. If i'm understanding how that works correctly.
![image](https://user-images.githubusercontent.com/47460175/122612009-f8c75300-d047-11eb-96a9-f1ed4411f825.png)

**Reply by viceice** [reply]
Can you please verify by running `renovate --version` ?

Maybe the docker season has cached the image from last build. This happens when you don't explicit pull the base image before building. Or do you pin and update docker tag or digest?

**Reply by dbimhoff** [reply]
just ran and gave me 25.41.3

**Reply by rarkins** [reply]
Click the retry/rebase checkbox in the MR and then make sure to capture debug logs the next time it runs on that repo

**Reply by dbimhoff** [reply]
I'm not sure why but I guess checking the box and rerunning it seems to have fixed the pipeline? I wish I could provide better input on to why this happened and how I fixed it, but I really am not sure.
